src/components/checkBoxSelection/checkBoxSelection.js

```diff
@@ -1,47 +1,112 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import './checkBoxSelection.scss';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "./checkBoxSelection.scss";
+import { connect } from "react-redux";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class CheckBoxSelection extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: false
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected
+    };
+  }
+
+  convertSvgToDataUrl = file => {
+    let reader = new FileReader();
+    reader.onload = result => {
+      this.props.svgSettingList.push({
+        name: file.name,
+        dataurl: result.target.result
+      });
+    };
+    if (file) {
+      reader.readAsDataURL(file);
     }
+  };
 
-    componentWillMount() {
-	 this.setState({ isClicked: this.props.isSelected }); 
+  changeColourAndTick = () => {
+    let isFound = false;
+    let index = 0;
+    this.setState({ isClicked: !this.state.isClicked });
+    for (var key in this.props.svgSettingList) {
+      if (
+        this.props.svgSettingList[key].name === this.props.file.name
+      ) {
+        isFound = true;
+        index = key;
+      }
     }
+    if (!isFound) {
+      this.convertSvgToDataUrl(this.props.file);
+    } else {
+      this.props.svgSettingList.splice(index, 1);
+      isFound = false;
+    }
+  };
 
-	handleDivClick= () => {
-	    this.setState({
-	        isClicked: !this.state.isClicked
-	    });
-	}
-
-	render() {
-
-	    return (
-
-	        <div
-    		className={this.state.isClicked ? 'checkbox-container-clicked' : 'checkbox-container-unclicked'}
-    		onClick={this.handleDivClick}
-	        >
-	            <input
-    			className='checkbox-input'
-    			type='checkbox'
-    			checked={this.state.isClicked}
-	            />
-	            <label className='checkbox-label'>
-					  {this.props.filename}
-	            </label>
-	        </div>
-	    );
-	}
+  render() {
+    return (
+      <div
+        className={
+          this.state.isClicked
+            ? "checkbox-container-clicked"
+            : "checkbox-container-unclicked"
+        }
+        onClick={this.changeColourAndTick}
+      >
+        <input
+          className="checkbox-input"
+          type="checkbox"
+          checked={this.state.isClicked}
+        />
+        <label className="checkbox-label">{this.props.filename}</label>
+      </div>
+    );
+  }
 }
 
 CheckBoxSelection.propTypes = {
-    filename: PropTypes.string,
-    isSelected: PropTypes.bool
+  svgSettingList: PropTypes.object,
+  svgSettingList : PropTypes.object,
+  svgOptions: PropTypes.object,
```

Comment:
prop not used, please go through all your files and make sure you are not importing unused props

Reply:
changed it.